### PR TITLE
Module 3 hw

### PR DESCRIPTION
**Pull request by malakhau**
The tests passed for every task.

### CODE REVIEW

File: module-3/task1/pointers/pointers_test.go [hunk 1]
```diff
@@ -14,9 +14,9 @@ func Test_doubleValues(t *testing.T) {
 	}
 
 	for i, tt := range tests {
-		out, expected := *tt.in, *tt.expected
-		if DoubleValues(tt.in); expected == out {
-			t.Errorf("Unexpected result in Test #%d, expected %+v, got %+v", i, tt.expected, *tt.in)
+		out, expected := &tt.in, &tt.expected
+		if DoubleValues(out); expected == out {
+			t.Errorf("Unexpected result in Test #%d, expected %+v, got %+v", i, tt.expected, tt.in)
```
Comment (wshaman) [inline module-3/task1/pointers/pointers_test.go:19]: Will `tt.in` on this row contain the result value?
Why?
In fact, it's more about readability - usually you want to see a variable named NOT `in` as a value for `got`

**Review by wshaman — APPROVED** [review]
:+1:

File: module-3/task2/slices/slice.go [hunk 2]
```diff
@@ -13,9 +19,15 @@ type User struct {
 }
 
 func SortUsersPure(s []User) []User {
+	r := make([]User, len(s))
+	copy(r, s)
+	sort.Slice(r, func(i, j int) bool { return r[i].FirstName+r[i].LastName < r[j].FirstName+r[j].LastName })
```
Comment (wshaman) [inline module-3/task2/slices/slice.go:24]: sweet solution. But one case can break it:
`aaa ab`
`aa abb`

Comment (malakhau) [reply module-3/task2/slices/slice.go:24]: SortUsersPure is used to sort []User, which contains FirstName and LastName so I don't think that your case is possible here.

**Pull request merged**